### ISSUE TITLE
Module 'tensorflow.keras' has no attribute '__version__' when running examples

### ISSUE THREAD

**Issue by w568w** [thaler-lab/EnergyFlow#29]
An error was encountered while running the [EFN example](https://github.com/pkomiske/EnergyFlow/blob/master/examples/efn_example.py):

```
/home/w568w/Projects/efn/.venv/lib/python3.11/site-packages/energyflow/archs/__init__.py:30: UserWarning: could not import some architectures - cannot import name '__version__' from 'tensorflow.keras' (/home/w568w/Projects/efn/.venv/lib/python3.11/site-packages/keras/api/_v2/keras/__init__.py) 
  warnings.warn('could not import some architectures - ' + str(e)) 
Traceback (most recent call last): 
  File "/home/w568w/Projects/efn/efn_example.py", line 32, in <module> 
    from energyflow.archs import EFN 
ImportError: cannot import name 'EFN' from 'energyflow.archs' (/home/w568w/Projects/efn/.venv/lib/python3.11/site-packages/energyflow/archs/__init__.py)
```

After inspection, I found that the module `tensorflow.keras` has been [splited into a separate package](https://github.com/tensorflow/tensorflow/commit/2e6acf753381cb7902017bcab03e204bb41b7091#diff-9515545cab2c345b253744198fdfdbe2aa6b3bfd8eb1fbb4abb600e804561191) since [2.6.0 released](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md#release-260), 2 yrs ago. `tensorflow.keras` still exists as a backward-compatible API endpoint. **But of course, the basic attribute of a package, `__version__`, does not come along with the export.**

However, EnergyFlow is still looking for:

https://github.com/pkomiske/EnergyFlow/blob/e5eb2bdacba93fb933f63185817df7f37ad32b58/energyflow/archs/efn.py#L18

which is incompatible on any TensorFlow version greater than 2.6.0.

## Reproducing Steps
1. Create a clean environment:
```shell
$ mkdir efn
$ cd efn
$ PIPENV_VENV_IN_PROJECT=1 pipenv shell
(efn) $ pip install tensorflow scikit-learn matplotlib EnergyFlow
```
2. Put `efn_example.py` into the directory;
3. Run `(efn) $ python efn_example.py`;
4. The error happens.

## Environment Info
```
OS: Archlinux (Rolling Release)
Python: 3.11.3
pipenv: version 2023.7.23
TensorFlow: 2.13.0
EnergyFlow: 1.3.2
```

**Comment by w568w** [comment]
BTW, `utils/data_utils.py` is using a method `np.int()`, [deprecated since version 1.20.0](https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations), almost 3 years ago:

https://github.com/pkomiske/EnergyFlow/blob/e5eb2bdacba93fb933f63185817df7f37ad32b58/energyflow/utils/data_utils.py#L176

In Numpy 1.24.3, this results in an `AttributeError: module 'numpy' has no attribute 'int'`. Please consider replacing it with one of `np.int_`, `np.int32` or `np.int64`.

**Comment by YuanchenZhou** [comment]
Hi, I am working on a project with @mattleblanc, and we are still having this issue. Is there any advice that we can apply? @matthewfeickert @rikab

**Comment by mattleblanc** [comment]
Indeed -- we can get it to work in a notebook by adding the following lines, but they don't fix the problem in `pfn_example.py` :

```
# to fix keras version problem with energyflow
# via https://stackoverflow.com/questions/76650327/how-to-fix-cannot-import-name-version-from-tensorflow-keras
from keras import __version__
tf.keras.__version__ = __version__
```

**Comment by rikab** [comment]
Hi, thanks for bringing this to our attention!

I will work on a hotfix and put it out shortly! But in the meantime, I have found that the following works as a temporary solution:

>  I have found that a working solution is to, in a clean environment, install only tensorflow 2.12.0, but without manually installing Keras. Then, install energyflow afterwards. I haven't tested if this also works for 2.13.0. Keras should be automatically installed alongside tensorflow -- this issue occurs because of Keras version number problems, as there are parts of the code that explicitly depend on the version number. My packages look like this:

> EnergyFlow                    1.3.2
> numpy                         1.22.3
> keras                         2.12.0
> tensorflow                    2.12.0
> pip                           23.1.2
> scikit-learn                  1.2.2
> scipy             1.10.1

**Comment by mattleblanc** [comment]
Awesome, thanks @rikab! We are able to develop a bit with this -- agreed that it will probably be best to ultimately fix it within `energyflow`.

**Comment by matthewfeickert** [comment]
@YuanchenZhou I can respond in a bit to describe a workflow that is reproducible for use in the future, but can you please provide environment information? Specifically:

* What is your OS and architecture?
* What version of EnergyFlow are you using?
* What version of TensorFlow are you using?

e.g. what is

```
python -m pip list | grep 'EnergyFlow\|tensorflow'
```

**Comment by YuanchenZhou** [comment]
@matthewfeickert 
I am using Linux (RHEL, version 9.2)
Architecture x86_64
Python 3.9.16
EnergyFlow 1.3.2
TensorFlow 2.16.1

**Comment by rikab** [comment]
Should be resolved by v1.3.3!
